iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/string_internal.hpp

```diff
@@ -113,6 +119,7 @@ struct GetData<string<N>>
 };
 
 template <uint64_t N>
+// NOLINTNEXTLINE(hicpp-avoid-c-arrays, cppcoreguidelines-avoid-c-arrays) cxx::string wraps char array
```

Comment:
I would state, provides uniform and safe access (in combination with `GetCapa` and `GetSize`) to string like constructs like `std::string`, string literal, char and `cxx::string`